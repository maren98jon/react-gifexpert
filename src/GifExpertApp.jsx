import { useState } from "react";
import { AddCategory, GifGrid } from "./componentes";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Formula 1']);

    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)) return;
        
        setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>GifExpertApp by __MarenDev__</h1>

            <AddCategory
                //setCategories={setCategories}
                onNewCategory={value => onAddCategory(value)}
            />

            {
                categories.map((category) => (
                    <GifGrid 
                        key={category} 
                        category={category}
                    />
                ))
            }
        </>
    )
}
