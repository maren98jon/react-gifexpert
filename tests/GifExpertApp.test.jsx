import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas en el <GifExpertApp/>', () => {

    test('debe de hacer match con el snapshot', () => { 

        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();

     });

     test('onAddCategory debe anyadir una categoria al principio',()=>{

        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        const inputValue = 'Argentina';

        fireEvent.input(input, { target: {value: inputValue}});
        fireEvent.submit(form);

        expect(screen.getAllByText(inputValue));
     })

 });