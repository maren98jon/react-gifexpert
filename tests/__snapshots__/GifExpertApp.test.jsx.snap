// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp by __MarenDev__
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar GIFs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Formula 1
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
